packages/echo-base/src/errors/BaseError.test.ts

```diff
@@ -0,0 +1,68 @@
+import { BaseError, getAllProperties } from './BaseError';
+
+describe('BaseError', () => {
+    const message = 'this is a test';
+    it('should initialize with correct values', () => {
+        const actualError = new BaseError({ name: 'BaseError', message });
+        expect(actualError.message).toBe(message);
+        expect(actualError.name).toBe('BaseError');
+        expect(actualError.stack).toBeTruthy();
```

Comment:
There is a less technical assertion for this one: `.toBeDefined()`. Maybe it better describes what we want here, then `.toBeTruthy()`

Reply:
I don't think toBeDefined works, since it will accept empty string.

.toBeTruthy() doesn't accept empty string
Use when you don't care what a value is, you just want to ensure a value is true in a boolean context. In JavaScript, there are six falsy values: false, 0, '', null, undefined, and NaN. Everything else is truthy.

Reply:
Ah, I see! I overlooked this one :,D My bad.